### PR TITLE
Median and Skipping rels supported

### PR DESCRIPTION
**Pull request by sungeunbae**
A cybershake-like batch of GMSim has a realisation selection file typically looks like:

```
AhuririR 30r
AkaOtaki 33r
Akatore 33r 
Albury 21r
```
However, some cybershake batches (eg. v24p8 , v24p9) include additional realisations to an existing batch. In such a case, the realisation selection file will looks like:
```
AhuririR 40 30r
```
This means we will skip the first 30 realisations for AhuririR, and run realisation 31, 32,... 40. 
The optional third column means the number of realisations to skip.

This PR also include a support for "SAMPLE_BB" type. We sometimes have a need to quickly inspect a BB file. In the past, BB files across all realisations are packed into a single tar file (sometimes multiple tar files), which enforced downloading a large-sized TAR file just to extract one for an inspection. The new inclusion of "SAMPLE_BB" type enables a separate upload of a sample event (median if available or the first realisation otherwirse) alongside the fully-packed BB tar file(s) to efficiently handle such a need.

When `cs_dropbox_preproess.py` is executed with `SAMPLE_BB` enabled, the `files_to_sync.yaml` file will have this entry:
```
   39   SAMPLE_BB:
   40     /scale_wlg_nobackup/filesets/nobackup/nesi00213/RunFolder/Cybershake/v24p8/Runs/AhuririR/AhuririR/BB/Acc/BB.bin: 1158716016
```

`cs_dropbox_upload.py` will be given this file `files_to_sync.yaml`, and uploads `SAMPLE_BB` separately alongside other files.

image

### CODE REVIEW

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_upload.py [hunk 1]
```diff
@@ -146,23 +189,45 @@ def copy_files(fault_name, data_type, work_dir):
 
 
 class TARGroup:
-    # Used to handle a group of TAR files when files to be archived are need to be partitioned due to upload size limit
-    # If a tar file, such as AlpineK2T_BB.tar is going to be larger than size limit (100Gb currently), we will partition files
-    # such that the resulting tar files below the size limit.
-    # As a result, we have AlpineK2T_BB_0.tar, AlpineK2T_BB_2.tar ..., with the last one ending with "f", such as AlpineK2T_BB_3f.tar
-    # This way, we know we have a complete set of TAR files if we find all 0, 1 ..3 labeled files with the AlpineK2T_BB prefix.
+    """
+    Used to handle a group of TAR files when files to be archived are need to be partitioned due to upload size limit
```
Comment (AndrewRidden-Harper) [inline dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_upload.py]:
```suggestion
    Used to handle a group of TAR files when files to be archived need to be partitioned due to upload size limit
```

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_upload.py [hunk 2]
```diff
@@ -146,23 +189,45 @@ def copy_files(fault_name, data_type, work_dir):
...
+
+    If a tar file, such as AlpineK2T_BB.tar is going to be larger than size limit (100Gb currently), we will partition files
+    such that the resulting tar files below the size limit.
```
Comment (AndrewRidden-Harper) [inline dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_upload.py]:
```suggestion
    such that the resulting tar files are below the size limit.
```

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_upload.py [hunk 3]
```diff
@@ -192,7 +278,7 @@ def __make_partition(all_files):
             one_partition.append(f)
             partition_size += f.stat().st_size
         else:  # including this will exceed. Finalize the current subset, start a new one
-            # TODO: f is assumned to be smaller than the size limit
+            # TODO: Currently f is assumed to be smaller than the size limit. Fix this if this assumption is not valid
             partition_list.append(one_partition)
             # print(one_partition)
```
Comment (AndrewRidden-Harper) [inline dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_upload.py]: delete these commented print() lines?

**Review by AndrewRidden-Harper — APPROVED** [review]
looks ok to me. Just a couple of suggestions for docstrings

File: dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_upload.py [hunk 4]
```diff
@@ -515,27 +691,27 @@ def update_uploaded_status(tar_files, uploaded_tar_dict, log=True, debug=True):
...
         for data_type in data_types:
-            if data_type not in uploaded_tar[fault_name]:
+            if data_type not in uploaded_files[fault_name]:
                 print(f"!!! {fault_name}_{data_type}*.tar had an uploading issue")
                 error_count += 1
 
     # verify if the supposely uploaded files indeed exist
     tar_files_found = retrieve_dropbox_files(dropbox_path, check_tar=True, debug=False)
-    uploaded_tar_2 = {}
+    uploaded_files_2 = {}
```
Comment (joelridden) [inline dropbox_rclone/dropbox_rclone/scripts/cs_dropbox_upload.py]: Does this need the 2? Or at least maybe a different variable name

File: dropbox_rclone/dropbox_rclone/scripts/files_to_sync.yaml
Comment (joelridden) [inline dropbox_rclone/dropbox_rclone/scripts/files_to_sync.yaml:1]: Is this file meant to be here?

**Review by joelridden — APPROVED** [review]

**Pull request merged**